### PR TITLE
Update Pub/Sub docs section

### PR DESCRIPTION
**Pull request by lukas-unity**
**What this PR does / why we need it**: Updates documentation to include section on how to grant access to Writer identity service account to publish messages from the log sink to the pub/sub topic.

also: fixes the `gcloud pubsub subscriptions create` command as it is not working at the moment due to invalid arguments set. 

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [X] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

File: docs/sources/clients/promtail/gcplog-cloud.md [hunk 1]
```diff
@@ -46,6 +46,34 @@ For more information on adding `log-filter` refer this [document](https://cloud.
...
+
+```bash
+$ gcloud pubsub topics add-iam-policy-binding $TOPIC_ID \
+--member=$WRITER_IDENTITY --role=roles/pubsub.publisher
+```
+
+e.g.:
+```bash
+$ gcloud pubsub topics add-iam-policy-binding cloud-logs \
+--member=serviceAccount:p97323568341-442148@gcp-sa-logging.iam.gserviceaccount.com --role=roles/pubsub.publisher
```
Comment (kavirajk) [inline docs/sources/clients/promtail/gcplog-cloud.md:74]: Hope it's not the real SA account name? :)

**Review by kavirajk — APPROVED** [review]
LGTM

**Comment by kavirajk** [comment]
@KMiller-Grafana can you take a final look? thanks

File: docs/sources/clients/promtail/gcplog-cloud.md (hunk 1 above)
Comment (lukas-unity) [reply docs/sources/clients/promtail/gcplog-cloud.md:74]: It is not :) we can probably replace these digits with x's if it makes things more clear

Comment (KMiller-Grafana) [inline docs/sources/clients/promtail/gcplog-cloud.md:74]:
```suggestion
## Grant log sink the pubsub publisher role

Find the writer identity service account of the log sink just created:

```bash
gcloud logging sinks describe \
 --format='value(writerIdentity)' $SINK_NAME
```

For example:
```bash
$ gcloud logging sinks describe \
 --format='value(writerIdentity)' cloud-logs
```

Create an IAM policy binding to allow log sink to publish messages to the topic:

```bash
gcloud pubsub topics add-iam-policy-binding $TOPIC_ID \
--member=$WRITER_IDENTITY --role=roles/pubsub.publisher
```

For example:
```bash
$ gcloud pubsub topics add-iam-policy-binding cloud-logs \
--member=serviceAccount:p987654321-123456@gcp-sa-logging.iam.gserviceaccount.com --role=roles/pubsub.publisher
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]
I've only made corrections on the added prose. The entire section is in great need of an edit. We'll address that edit in another PR.

**Comment by kavirajk** [comment]
@lukas-unity can you update the changes suggested by @KMiller-Grafana?

File: docs/sources/clients/promtail/gcplog-cloud.md (hunk 1 above)
Comment (owen-d) [reply docs/sources/clients/promtail/gcplog-cloud.md:74]: Yeah, I think that'd be best, even if it just prevents us from other issues being opened about this in the future 😆

**Review by owen-d — APPROVED** [review]

**Comment by lukas-unity** [comment]
Hey, sorry I was away for a bit, PR is ready for review now.

File: docs/sources/clients/promtail/gcplog-cloud.md [hunk 2]
```diff
@@ -46,6 +46,33 @@ For more information on adding `log-filter` refer this [document](https://cloud.
...
+
+Create an IAM policy binding to allow log sink to publish messages to the topic:
+```bash
+gcloud pubsub topics add-iam-policy-binding $TOPIC_ID \
+--member=$WRITER_IDENTITY --role=roles/pubsub.publisher
+```
+
+For example:
+```bash
+$ gcloud pubsub topics add-iam-policy-binding cloud-logs \
```
Comment (KMiller-Grafana) [inline docs/sources/clients/promtail/gcplog-cloud.md:72]:
```suggestion
gcloud pubsub topics add-iam-policy-binding cloud-logs \
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]
Small changes requested to remove the "$" prompt shown in the examples.

**Comment by lukas-unity** [comment]
Hey @KMiller-Grafana, suggestions applied, thank you!

**Review by KMiller-Grafana — APPROVED** [review]

**Comment by cstyan** [comment]
hey @lukas-unity it looks like the CI failures are unrelated to your changes, can you try merging master into your branch and see if that fixes the build for your PR?

**Comment by lukas-unity** [comment]
> hey @lukas-unity it looks like the CI failures are unrelated to your changes, can you try merging master into your branch and see if that fixes the build for your PR?

@cstyan done, all checks have passed!

**Comment by kavirajk** [comment]
Thanks @lukas-unity 👍

**Pull request merged**